Switch `to_timedelta` argument type from `list` to `Sequence`

**Describe the bug**
`to_timedelta` uses a list of a union argument type.
This will raise false positive errors when passing a list of a single type because list is invariant.

**To Reproduce**
```python
import pandas as pd

pd.to_timedelta(["1 day"])
```

produces:
> error: Argument 1 to "to_timedelta" has incompatible type "list[str]"; expected "Union[Sequence[Union[float, timedelta]], list[Union[str, float, timedelta]], tuple[Union[str, float, timedelta], ...], range, Union[ExtensionArray, ndarray[Any, Any]], Index[Any]]"  [arg-type]
> note: "List" is invariant -- see https://mypy.readthedocs.io/en/stable/common_issues.html#variance
> note: Consider using "Sequence" instead, which is covariant

**Please complete the following information:**
 - OS: Win
 - OS Version 10
 - python 3.9.7
 - version of type checker mypy 1.8.0
 - version of installed `pandas-stubs` 2.2.2.240603

--- a/pandas-stubs/core/tools/timedeltas.pyi
+++ b/pandas-stubs/core/tools/timedeltas.pyi
@@ -32,7 +32,6 @@ def to_timedelta(
 def to_timedelta(
     arg: (
         Sequence[str | float | timedelta]
-        | list[str | float | timedelta]
         | tuple[str | float | timedelta, ...]
         | range
         | ArrayLike

--- a/pandas-stubs/core/tools/timedeltas.pyi
+++ b/pandas-stubs/core/tools/timedeltas.pyi
@@ -31,7 +31,7 @@ def to_timedelta(
 @overload
 def to_timedelta(
     arg: (
-        Sequence[float | timedelta]
+        Sequence[str | float | timedelta]
         | list[str | float | timedelta]
         | tuple[str | float | timedelta, ...]
         | range

--- a/pandas-stubs/core/tools/timedeltas.pyi
+++ b/pandas-stubs/core/tools/timedeltas.pyi
@@ -32,7 +32,7 @@ def to_timedelta(
 @overload
 def to_timedelta(
     arg: (
-        SequenceNotStr[str]
+        SequenceNotStr
         | Sequence[float | timedelta]
         | tuple[str | float | timedelta, ...]
         | range

--- a/pandas-stubs/core/tools/timedeltas.pyi
+++ b/pandas-stubs/core/tools/timedeltas.pyi
@@ -14,6 +14,7 @@ from pandas._libs.tslibs.timedeltas import TimeDeltaUnitChoices
 from pandas._typing import (
     ArrayLike,
     RaiseCoerce,
+    SequenceNotStr,
 )
 
 @overload
@@ -31,7 +32,8 @@ def to_timedelta(
 @overload
 def to_timedelta(
     arg: (
-        Sequence[str | float | timedelta]
+        SequenceNotStr[str]
+        | Sequence[float | timedelta]
         | tuple[str | float | timedelta, ...]
         | range
         | ArrayLike

--- a/tests/test_timefuncs.py
+++ b/tests/test_timefuncs.py
@@ -25,7 +25,10 @@ if TYPE_CHECKING:
     from pandas._typing import FulldatetimeDict
 else:
     FulldatetimeDict = Any
-from pandas._typing import TimeUnit
+from pandas._typing import (
+    SequenceNotStr,
+    TimeUnit,
+)
 
 from tests import (
     PD_LTE_22,
@@ -141,7 +144,8 @@ def test_timedelta_series_arithmetic() -> None:
 
 
 def test_timedelta_series_string() -> None:
-    check(assert_type(pd.to_timedelta(["1 day"]), pd.TimedeltaIndex), pd.TimedeltaIndex)
+    seq_list: SequenceNotStr[str] = ["1 day"]
+    check(assert_type(pd.to_timedelta(seq_list), pd.TimedeltaIndex), pd.TimedeltaIndex)
 
 
 def test_timestamp_timedelta_series_arithmetic() -> None:

--- a/tests/test_timefuncs.py
+++ b/tests/test_timefuncs.py
@@ -140,6 +140,15 @@ def test_timedelta_series_arithmetic() -> None:
     r4: pd.TimedeltaIndex = tds1 / 10.2
 
 
+def test_timedelta_series_string() -> None:
+    tds1: pd.TimedeltaIndex = pd.to_timedelta(["1 day"])
+    td1: pd.Timedelta = pd.Timedelta("2 days")
+    r1: pd.TimedeltaIndex = tds1 + td1
+    r2: pd.TimedeltaIndex = tds1 - td1
+    r3: pd.TimedeltaIndex = tds1 * 4.3
+    r4: pd.TimedeltaIndex = tds1 / 10.2
+
+
 def test_timestamp_timedelta_series_arithmetic() -> None:
     ts1 = pd.to_datetime(pd.Series(["2022-03-05", "2022-03-06"]))
     assert isinstance(ts1.iloc[0], pd.Timestamp)

--- a/tests/test_timefuncs.py
+++ b/tests/test_timefuncs.py
@@ -25,10 +25,7 @@ if TYPE_CHECKING:
     from pandas._typing import FulldatetimeDict
 else:
     FulldatetimeDict = Any
-from pandas._typing import (
-    SequenceNotStr,
-    TimeUnit,
-)
+from pandas._typing import TimeUnit
 
 from tests import (
     PD_LTE_22,

--- a/tests/test_timefuncs.py
+++ b/tests/test_timefuncs.py
@@ -144,7 +144,7 @@ def test_timedelta_series_arithmetic() -> None:
 
 
 def test_timedelta_series_string() -> None:
-    seq_list: SequenceNotStr[str] = ["1 day"]
+    seq_list = ["1 day"]
     check(assert_type(pd.to_timedelta(seq_list), pd.TimedeltaIndex), pd.TimedeltaIndex)
 
 

--- a/tests/test_timefuncs.py
+++ b/tests/test_timefuncs.py
@@ -141,12 +141,7 @@ def test_timedelta_series_arithmetic() -> None:
 
 
 def test_timedelta_series_string() -> None:
-    tds1: pd.TimedeltaIndex = pd.to_timedelta(["1 day"])
-    td1: pd.Timedelta = pd.Timedelta("2 days")
-    r1: pd.TimedeltaIndex = tds1 + td1
-    r2: pd.TimedeltaIndex = tds1 - td1
-    r3: pd.TimedeltaIndex = tds1 * 4.3
-    r4: pd.TimedeltaIndex = tds1 / 10.2
+    check(assert_type(pd.to_timedelta(["1 day"]), pd.TimedeltaIndex), pd.TimedeltaIndex)
 
 
 def test_timestamp_timedelta_series_arithmetic() -> None:

